### ISSUE TITLE
Possible bug in lr scheduler

### ISSUE THREAD

**Issue by J-L-O** [microsoft/ClimaX#25]
Hi, thank you for releasing the code for your paper, it's been a great help for our own research. When finetuning the pretrained models using your code we noticed that the lr scheduler was behaving unexpectedly, and after some search we believe the reason for that lies in this code snippet:

https://github.com/microsoft/ClimaX/blob/efd6de4b7e3f0e552693ade5c90d250b6ef0b4b6/src/climax/global_forecast/module.py#L223-L230

It seems to me that the interval should be set as "epoch" and not "step" in line 230, as the milestones for the scheduler are also given in epochs.

**Comment by tung-nd** [comment]
Hi, thank you for your interest in ClimaX. The performance we got in the paper was by using a step-level scheduler, and to do that correctly we had to set the warmup_epochs and max_epochs to the number of warmup steps and max steps, respectively. To use an epoch-level scheduler then yes, you should modify line 230 to "interval": "epoch"

**Comment by tung-nd** [comment]
did your problem get resolved @J-L-O?

**Comment by rejuvyesh** [comment]
To clarify, we could have better variable names, but the scheduler implementation is correct.

**Comment by J-L-O** [comment]
Ah I see, that makes sense. Thank you for taking the time to answer!
